### PR TITLE
Add versioning info to filing instruction guide pages

### PR DESCRIPTION
**Pull request by contolini**
Filing instruction guide pages have statuses and effective dates that are shown to the user via a banner at the top of the page and a pill in the left-hand nav.

The banner and pill are only shown if the page status is anything other than `current`. The pill uses our "tag" pattern and modifier classes have been added to `tag.less`.

https://github.local/CFPB/1071-data-collection/issues/178
https://github.local/CFPB/1071-data-collection/issues/185

## Additions

- `__info` and `__dark` modifier classes to our tag design pattern.
- Banner and pill (tag) to FIG pages if the page's `version_status` is set.
- Version 0.17 of the DS

## Changes

- Moves the FIG template logic out of `layout-side-nav.html` and into it's own template file because the versioning additions are unique to FIG.

## How to test this PR

1.  Check out the [previous](https://demo-server-4.cfpb.gov/compliance/compliance-resources/small-business-lending/1071-filing-instruction-guide-out-of-date/) and [archived](https://demo-server-4.cfpb.gov/compliance/compliance-resources/small-business-lending/1071-filing-instruction-guide-archived/) example pages on our dev server.

## Screenshots

| previous version | previous version | archived version | archived version |
|--------|-------|---|---|
| image | image | image | image |

## Checklist

- [ ] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentially one or more of:
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18 (the last Edge prior to it switching to Chromium)
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

File: cfgov/filing_instruction_guide/jinja2/filing_instruction_guide/layout-fig.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            <div class="wrapper
+                        wrapper__match-content">
+                {{ notification.render(
+                    'warning',
+                    true,
+                    version_message,
+                    date_message,
+                    [ {
+                        'text': 'View all versions of the FIG.',
+                        'url': '#'
```
Comment (schbetsy) [inline cfgov/filing_instruction_guide/jinja2/filing_instruction_guide/layout-fig.html:31]: I think (?) the index of all versions will probably be one level up from the FIG page itself, but I'm not sure. In any case, we should put a TODO around this and create a ticket so we remember to update it.

```suggestion
                        'url': '../'
```

**Review by schbetsy — COMMENTED** [review]
Looks good and works well, including on mobile.

Chris, one issue I'm interested in your thoughts on: I think eventually we want to be able to use this page (with the sticky table of contents, etc.) for other kinds of content besides FIG stuff. BUT, there isn't a specific use case yet, so making it reusable is only valuable in theory. 

This PR adds FIG-specific strings, like "This version is not the current FIG", so if anyone does want to re-use this page type, they'd have to do some dev work to extract that from the template and make it generic. Do we... care? I'm on the fence.

File: cfgov/filing_instruction_guide/jinja2/filing_instruction_guide/layout-fig.html (hunk 1 above)
Comment (contolini) [reply cfgov/filing_instruction_guide/jinja2/filing_instruction_guide/layout-fig.html:31]: Another thought: maybe this should be defined in the CMS? A field in wagtail that lets the user input the URL of the index/history page in the event we don't list all the versions on the FIG landing page.

**Comment by contolini** [comment]
> This PR adds FIG-specific strings, like "This version is not the current FIG", so if anyone does want to re-use this page type, they'd have to do some dev work to extract that from the template and make it generic. Do we... care? I'm on the fence.

I had the exact same thought process and ultimately decided to avoid premature generalization and appended `-fig` to the files to avoid confusion. I like the idea of using our TOC technique on other pages but right now its tightly coupled to the [FIG model](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/filing_instruction_guide/models.py#L118-L135). Without the auto-generated FIG section IDs the sidenav would remain collapsed and not track the contents of the page.

Sooooo I agree that this functionality should be generalized for other use but I'd vote to have two use cases before we begin that abstraction. E.g. we could use JS to read all the h2/h3/h4/h5s on the page and assign IDs and TOC sub-nav items on page load. Or we could create a new TOC page type and do a technique similar to what we do with FIG pages. Currently, the left-hand nav of cf.gov is used to link to sibling pages so any page with a TOC would have to override that functionality.

@wpears started a similar conversation in #off-topic so we should definitely continue discussing this in a github issue somewhere.

**Review by schbetsy — APPROVED** [review]
(We already discussed this in a meeting, but also writing it here for posterity:) This looks good for our FIG purposes, so I think we should merge. We'll start a discussion about what other use cases there might be for a page with a table of contents, and do any further development (to make the page work for non-FIG content) once we know what requirements those pages might have.

**Pull request merged**